apps/webapp/src/state/staking.ts

```diff
@@ -176,11 +180,8 @@ export const createStakingSlice = (): SliceCreator<StakingSlice> => (set, get) =
     });
   },
   onSubmit: () => {
-    /**
-     * @todo: Add switching logic depending on the value of `action`, once
-     * undelegation is implemented.
-     */
-    void get().staking.delegate();
+    if (get().staking.action === 'delegate') void get().staking.delegate();
+    else void get().staking.undelegate();
```

Comment:
Think the buttons should just call `delegate()` / `undelegate()` fns directly

Reply:
[Done](https://github.com/penumbra-zone/web/pull/651/commits/0e2b8edca18ae9b9da92121f8678157ef3e81a7b)